tests/rptest/tests/pandaproxy_test.py

```diff
@@ -2129,3 +2131,103 @@ def test_upgrade_and_enable_basic_auth(self, base_release: tuple[int, int],
+        self.logger.info("Create a consumer group")
+        group_id = f"pandaproxy-group-{uuid.uuid4()}"
+        cc_res = self._create_consumer(group_id)
+        assert cc_res.status_code == requests.codes.ok
+
+        c0 = Consumer(cc_res.json(), self.logger)
+
+        # Subscribe a consumer to topic
+        self.logger.info(f"Subscribe consumer to topic: {self.topic}")
+        sc_res = c0.subscribe([self.topic])
+        assert sc_res.status_code == requests.codes.no_content
+
+        # Attempt to read from topic
+        self.logger.info(f"Consumer fetch")
+        cf_res = c0.fetch(headers=HTTP_CONSUMER_FETCH_JSON_V2_HEADERS)
+        assert cf_res.status_code == requests.codes.ok, f"Status: {cf_res.status_code}"
+        fetch_result = cf_res.json()
+        self.logger.debug(f"fetch result {json.dumps(fetch_result)}")
+        assert len(fetch_result) == 3
+
+        def do_consumer_offset_commit(new_offset: int):
+            sco_req = dict(partitions=[
+                dict(topic=self.topic, partition=0, offset=new_offset)
+            ])
+            co_res_raw = c0.set_offsets(data=json.dumps(sco_req))
+            assert co_res_raw.status_code == requests.codes.no_content
+
+        def do_consumer_offset_fetch():
+            co_req = dict(partitions=[dict(topic=self.topic, partition=0)])
+            offset_result_raw = c0.get_offsets(data=json.dumps(co_req))
+            assert offset_result_raw.status_code == requests.codes.ok
+            res = offset_result_raw.json()
+            self.logger.debug(f"offset result {json.dumps(res)}")
+            # Should be one offset for each partition that we previously set offsets for
+            assert len(res["offsets"]) == 1
+            for r in res["offsets"]:
+                assert r["topic"] == self.topic
+                assert r["partition"] == 0
+                assert r["offset"] == 0
+                assert r["metadata"] == ""
+
+        def do_coordinator_change(utils: GroupCoordinatorTransferUtils):
+            new_leader = utils.select_next_leader()
+
+            wait_until(lambda: utils.transfer_leadership(new_leader),
+                       timeout_sec=10,
+                       backoff_sec=1)
+
+            wait_until(lambda: utils.validate_leadership_transfer(new_leader),
+                       timeout_sec=30,
+                       backoff_sec=5)
+
+        utils = GroupCoordinatorTransferUtils(self.redpanda)
+
+        do_consumer_offset_commit(new_offset=0)
+
+        self.logger.debug("Test offset fetch: before coordinator change")
+        do_consumer_offset_fetch()
+        do_coordinator_change(utils)
+        self.logger.debug("Test offset fetch: after coordinator change")
+        do_consumer_offset_fetch()
+
+        self.logger.debug("Change the coordinator again")
+        do_coordinator_change(utils)
+        self.logger.debug(
+            "Test offset commit: after second coordinator change")
+        do_consumer_offset_commit(new_offset=1)
```

Comment:
Is it worth checking that a fetch also works and gets expected results? (another PR makes sense for this if it doesn't work)

Reply:
+1

Reply:
It is not worth checking that a fetch also works. This is because `consumer::fetch` is executed on the leader for each topic-partition and **not** the consumer group coordinator. Therefore, that method would **not** test a coordinator change.
I wrote about that in bullet (2) of [this](https://github.com/redpanda-data/redpanda/pull/9967#issuecomment-1512278835) comment.

Unless you are asking about testing `consumer::offset_fetch`? In that case, I am already testing that in line 2227.

Reply:
Spoke offline about this. There is not need to test fetch here.